### PR TITLE
Ease using regipy as a library 

### LINKED ISSUE #171: logbook dep
Hi:
Logbook as a project seems like dead. Would you care for a PR to remove it and replace it with plain simpler logging?

### PR DESCRIPTION
**Pull request by pombredanne**
This PR resolves #171 and #172:
- #171 replace logbook by the standard library logging
- #172 make some deps optional with an extra_requires for [cli] to ease use as a library

It also removes the jsonlines dependency and uses instead the standard library `json` module to the same effect.

It makes also some minor extra changes such as:
- escaping a regex
- marking Python 3.6 and 3.9 as supported
- minor documentation adjustments

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

### CODE REVIEW

**Review by tincho9 — COMMENTED** [review]
Thanks, I really like this solution.

**Comment by pombredanne** [comment]
@tincho9 ready to merge IMHO :)

**Review by tincho9 — CHANGES_REQUESTED** [review]
I would like to move back the import to outside the plugin run function.

File: regipy/plugins/software/last_logon.py [hunk 1]
```diff
@@ -17,6 +15,8 @@ class LastLogonPlugin(Plugin):
     COMPATIBLE_HIVE = SOFTWARE_HIVE_TYPE
 
     def run(self):
+        from inflection import underscore
```
[the remark this answers is not in the document]
Comment (pombredanne) [reply regipy/plugins/software/last_logon.py:18]: I can move that out... that's used anyway. It has no impact here to have a method-private vs. a top level import.

[the remark this answers is not in the document]
Comment (tincho9) [reply regipy/plugins/software/last_logon.py]: I just think the code is clearer that way :)

**Review by tincho9 — APPROVED** [review]

File: regipy/plugins/software/last_logon.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (pombredanne) [reply regipy/plugins/software/last_logon.py]: The latest push restored the original imports

**Comment by tincho9** [comment]
Looks great now! I think we should bump the version to 2.0.0, since it is a bit of a breaking change for people that use the CLI. After that change we can merge.

**Comment by pombredanne** [comment]
> Looks great now! I think we should bump the version to 2.0.0, since it is a bit of a breaking change for people that use the CLI. After that change we can merge.

Alternatively there could be two packages built:
- `regipy` which has the same set of deps as today
-  `regipy-lib` which a minimal deps for library usage which is a bit more involved as the build scripts needs to be special to handle this

**Comment by pombredanne** [comment]
@tincho9 thank you ++ :bow:

**Pull request merged**